### DISCUSSION TITLE
pde_operator redundant points

### DISCUSSION THREAD

**Discussion by ddg93** [lululxvi/deepxde#1869] [Q&A]
Hello,
I am observing that when I create a deepxde.data.pde_operator.PDEOperator with num_function=1 and I had previously specified a PDE with, as an example, num_domain=100, num_boundary=20, num_initial=50, if I look into the pde_operator.train_bc[0] the size is correctly [50+20,N_sensors] but, instead, pde_operator.train_x[0] will have size [240,N_sensors], being 240=100+2*(20+50), i.e. twice the number of bc and ic points. Moreover, all of these are passed in the forward pass and PDE evaluation. What am I missing? Why are there twice the initial and boundary points?

**Comment by lululxvi** [comment]
Please read the `dde.data.PDE` class
